File: app/src/main/java/com/example/ogan/codedenim/ServiceGenerator.java
package com.example.ogan.codedenim;

import com.example.ogan.codedenim.gson.ApiMethods;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by belema on 9/11/17.
 */

public class ServiceGenerator {

    private static final String BASE_URL = "https://codedenim.azurewebsites.net/";


    private static Retrofit.Builder builder =
            new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create());

    private static Retrofit retrofit = builder.build();

    public static ApiMethods apiMethods = retrofit.create(ApiMethods.class);

    /*public ApiMethods getApiMethods(){
        return apiMethods;
    }

    public static <S> S createService(
            Class<S> serviceClass) {
        return retrofit.create(serviceClass);
    } */
}

File: app/src/main/java/com/example/ogan/codedenim/ServiceGenerator.kt
package com.example.ogan.codedenim

import com.example.ogan.codedenim.gson.ApiMethods
import com.google.gson.GsonBuilder
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit




/**
 * Created by belema on 9/11/17.
 */

object ServiceGenerator {

    private val BASE_URL = "https://codedenim.azurewebsites.net/"

    var okHttpClient = OkHttpClient.Builder()
            .connectTimeout(1, TimeUnit.MINUTES)
            .readTimeout(30, TimeUnit.SECONDS)
            .writeTimeout(15, TimeUnit.SECONDS)
            .build()

    var gson = GsonBuilder()
            .setLenient()
            .create()

    private val builder = Retrofit.Builder()
            .baseUrl(BASE_URL)
            .client(okHttpClient)
            .addConverterFactory(GsonConverterFactory.create(gson))

    private val retrofit = builder.build()

    val apiMethods = retrofit.create(ApiMethods::class.java)

    /*public ApiMethods getApiMethods(){
        return apiMethods;
    }

    public static <S> S createService(
            Class<S> serviceClass) {
        return retrofit.create(serviceClass);
    } */
}

